### PR TITLE
Add reachability check and better error message

### PR DESCRIPTION
**Pull request by haiyingweng**
Sorry it took me so long to finally put up this PR 😅

## Overview

Basically added Reachability pod to help check for internet connection. So home screen says no internet if there is no internet. Any problem connecting to backend would say could not connect to server 

## Changes Made

^

## Test Coverage

I tested running app on my phone. It's a little hard to cut off your internet with the simulator. 

## Related PRs or Issues (delete if not applicable)

#82 

## Screenshots (delete if not applicable)

<details>

  <summary>Internet connection stuff</summary>

ezgif-7-33a99212b573

  
  

</details>

### CODE REVIEW

File: CourseGrab/Network/Endpoints.swift [hunk 1]
```diff
@@ -15,7 +15,7 @@ extension Endpoint {
         Endpoint.config.keyDecodingStrategy = .convertFromSnakeCase
         Endpoint.config.keyEncodingStrategy = .convertToSnakeCase
 
-        #if LOCAL
+        #if DEBUG
```
Comment (mathewscullin) [inline CourseGrab/Network/Endpoints.swift:18]: Should we change this back to `#if LOCAL` or revert it back once backend deploys?

**Review by mathewscullin — APPROVED** [review]
Wonderful stuff! 💯

File: CourseGrab/Network/Endpoints.swift (hunk 1 above)
Comment (haiyingweng) [reply CourseGrab/Network/Endpoints.swift:18]: oh I'm not so sure what's the difference between LOCAL and DEBUG lol but I'll put it back to LOCAL

Comment (ywsang) [reply CourseGrab/Network/Endpoints.swift]: `LOCAL`/`DEBUG` refer to the build configuration that's being used when the project runs! we don't actually have a build configuration named `Local` (you can see them in when you click on just the project in Xcode sidebar in the "Info" tab), so the `#if LOCAL` is never actually hit lol -- if backend isn't deployed yet i think it's okay to keep it as `DEBUG` if you guys plan on continuing to use locally deployed backend to dev on

File: CourseGrab/ViewControllers/Home/HomeTableViewController.swift [hunk 2]
```diff
@@ -72,15 +74,41 @@ class HomeTableViewController: UITableViewController {
         show(state: .loading)
     }
 
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        NotificationCenter.default.addObserver(self, selector: #selector(reachabilityChanged(note:)), name: .reachabilityChanged, object: reachability)
+        do {
+            try reachability.startNotifier()
+        } catch {
+            print("could not start reachability notifier")
```
Comment (ywsang) [inline CourseGrab/ViewControllers/Home/HomeTableViewController.swift:83]: can we capitalize the first letter?

**Review by ywsang — APPROVED** [review]
🚀

**Pull request merged**